Catalog/Catalog.cpp

```diff
@@ -160,6 +160,7 @@ inline auto table_json_filepath(const std::string& base_path,
 }  // namespace
 
 Catalog::Catalog(const string& basePath,
+                 std::shared_ptr<ForeignStorageInterface> fsi,
```

Comment:
make sense. will do

Reply:
I found that 2 instances of the GlobalFileMgr are created: in the PersistentStorageMgr and ForeignStorageCache. This leads to FSI duplication which is not allowed. Is it done deliberately?

Reply:
ForeignStorageCache does not apply to FSI, and it will soon be updated to no longer use GlobalFileMgr. For now, we can ensure that the GlobalFileMgr instance used in ForeignStorageCache has its FSI member always set to nullptr.

Reply:
I made PersistentStorageMgr an owher of the FSI. Take a look please, perhaps this is an acceptable solution, at least while the ForeignStorageCache use GlobalFileMgr.